codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddSigv4aPlugin.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.aws.typescript.codegen;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import software.amazon.smithy.codegen.core.SymbolProvider;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.knowledge.ServiceIndex;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.Trait;
+import software.amazon.smithy.typescript.codegen.LanguageTarget;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.typescript.codegen.integration.TypeScriptIntegration;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Detects when to add sigv4a signer.
+ */
+@SmithyInternalApi
+public final class AddSigv4aPlugin implements TypeScriptIntegration {
+    public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
+        TypeScriptSettings settings,
+        Model model,
+        SymbolProvider symbolProvider,
+        LanguageTarget target
+    ) {
+        boolean useSigv4aSigner = false;
+
+        ServiceIndex serviceIndex = ServiceIndex.of(model);
+        Set<ServiceShape> services = model.getServiceShapes();
+        for (ServiceShape service of services) {
+            Map<ShapeId, Trait> authSchemes = serviceIndex.getAuthSchemes(service);
+            useSigv4aSigner = useSigv4aSigner || authSchemes.values().stream()
+                .anyMatch(trait -> trait.toString().equals("aws.auth#sigv4a"));
```

Comment:
I'm using a string because there doesn't seem to be a SigV4ATrait class yet

Reply:
The `SigV4ATrait` is available in the repo's smithy version 1.45.0: https://smithy.io/javadoc/1.45.0/software/amazon/smithy/aws/traits/auth/SigV4ATrait.html